scipp_widgets/widgets.py

```diff
@@ -188,3 +187,37 @@ def _process(self, kwargs):
         output = self.callable(**kwargs)
         self.scope[output_name] = output
         display(self.scope[output_name])
+
+
+class LoadWidget(WidgetBase):
+    """
+    Provides a simple graphical wrapper around a load function,
+    adding the return value to the notebooks scope labelled
+    by file name.
+    """
+    def __init__(self,
+                 wrapped_func: Callable,
+                 inputs: Iterable[IInput],
+                 button_name: str = 'Load',
+                 layout='row wrap',
+                 filename_param='filename',
+                 hide_code: bool = False):
+        """
+        :param filename_param: Name of input parameter to use as file name.
+        """
+        super().__init__(wrapped_func,
+                         inputs,
+                         button_name,
+                         hide_code=hide_code,
+                         layout=layout)
+        self.scope = get_notebook_global_scope()
+        self._filename_param = filename_param
+
+    def _process(self, kwargs):
+        """
+        Calls the wrapped function using the
+        parameter values specified.
+        """
+        output = self.callable(**kwargs)
+        self.scope[kwargs[self._filename_param]] = output
```

Comment:
Not sure if this is making trouble, but I could not get this to work. In the example from the docs `filename_param` is set to `value`, but if I try to access that after hitting `Load` it is not defined.